### PR TITLE
#33: support making both Prometheus.Summary and Prometheus.Histogram for com.ringcentral.platform.metrics.histogram.Histogram

### PR DESCRIPTION
**Pull request by devromik**
…

### CODE REVIEW

File: metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/samples/prometheus/PrometheusSampleMaker.java [hunk 1]
```diff
@@ -13,22 +13,66 @@
...
 import static java.util.stream.Collectors.toList;
 
 public class PrometheusSampleMaker implements SampleMaker<
     PrometheusSample,
     PrometheusSampleSpec,
-    PrometheusInstanceSampleSpec> {
+    PrometheusInstanceSampleSpec,
+    PrometheusInstanceSample> {
+
+    public static boolean DEFAULT_SEPARATE_HISTOGRAM_AND_SUMMARY = true;
```
Comment (mairovichaa) [inline metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/samples/prometheus/PrometheusSampleMaker.java:27]: should it be private and final?

Comment (devromik) [reply metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/samples/prometheus/PrometheusSampleMaker.java:27]: It should be final, of course. I'll fix it. 
It also should be public to be used in the builder class. Currently, it isn't used in the builder class and I'll fix it as well.
Thanks.

Comment (devromik) [reply metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/samples/prometheus/PrometheusSampleMaker.java]: Fixed.

File: metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/samples/prometheus/PrometheusSampleMaker.java [hunk 2]
```diff
@@ -70,35 +114,91 @@ public PrometheusSample makeSample(PrometheusSampleSpec spec, PrometheusInstance
...
+
+        MetricName childSampleChildInstanceSampleNameSuffix =
+            type == HISTOGRAM ?
+            summaryChildInstanceSampleNameSuffix :
+            histogramChildInstanceSampleNameSuffix;
+
+        Collector.Type childSampleChildInstanceSampleType = type == HISTOGRAM ? SUMMARY : HISTOGRAM;
+
+        if (m instanceof Count || m instanceof TotalSum) {
+            PrometheusSample childSample = new PrometheusSample(
```
[the remark this answers is not in the document]
Comment (devromik) [reply metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/samples/prometheus/PrometheusSampleMaker.java:161]: Good point! I'll implement the second approach.

[the remark this answers is not in the document]
Comment (devromik) [reply metrics-facade-prometheus/src/main/java/com/ringcentral/platform/metrics/samples/prometheus/PrometheusSampleMaker.java]: Done.

**Review by mairovichaa — APPROVED** [review]

**Pull request merged**